sys/include/net/conn/udp.h

```diff
@@ -93,6 +96,8 @@ int conn_udp_getlocaladdr(conn_udp_t *conn, void *addr, uint16_t *port);
  *
  * @note    Function may block.
  *
+ * @note    This function needs to be called from the same function as @ref conn_udp_create.
```

Comment:
I don't know what you mean by "using conn", but if I call `conn_udp_recvfrom()` (which is basically a wrapper for `msg_receive()`) from a different thread than the one that got registered, I'm unlikely to receive anything, right?